account_events.go

```diff
@@ -9,28 +10,129 @@ import (
 // Event represents an action taken on the Account.
 type Event struct {
 	CreatedStr string `json:"created"`
-	UpdatedStr string `json:"updated"`
 
-	ID              int
-	Status          string
-	Action          string
+	// The unique ID of this Event.
+	ID int
+
+	// Current status of the Event, Enum: "failed" "finished" "notification" "scheduled" "started"
+	Status EventStatus
+
+	// The action that caused this Event. New actions may be added in the future.
+	Action EventAction
+
+	// A percentage estimating the amount of time remaining for an Event. Returns null for notification events.
 	PercentComplete int `json:"percent_complete"`
-	Rate            string
-	Read            bool
-	Seen            bool
-	TimeRemaining   int
-	Username        string
-	Entity          *EventEntity
-	Created         *time.Time `json:"-"`
+
+	// The rate of completion of the Event. Only some Events will return rate; for example, migration and resize Events.
+	Rate string
+
+	// If this Event has been read.
+	Read bool
+
+	// If this Event has been seen.
+	Seen bool
+
+	// The estimated time remaining until the completion of this Event. This value is only returned for in-progress events.
+	TimeRemaining int
+
+	// The username of the User who caused the Event.
+	Username string
+
+	// Detailed information about the Event's entity, including ID, type, label, and URL used to access it.
+	Entity *EventEntity
+
+	// When this Event was created.
+	Created *time.Time `json:"-"`
 }
 
+// EventAction constants start with Action and include all known Linode API Event Actions.
+type EventAction string
+
+const (
```

Comment:
https://github.com/abice/go-enum is handy tool for generating enum for Go. We use this.

Reply:
thanks @tamalsaha will definitely check it out

Reply:
It's certainly interesting.  I think there is lots of room here to move towards generators.